utils/process/src/lib.rs

```diff
@@ -80,12 +100,21 @@ pub enum ExeUnitExitStatus {
 #[derive(Clone)]
 pub struct ProcessHandle {
     process: Arc<SharedChild>,
+    // Windows process will be killed once `job_object` is dropped
+    #[allow(unused)]
+    #[cfg(windows)]
+    job_object: JobObject,
```

Comment:
Having conditionally existing field is hard to maintain. I'm wondering if we could have empty job object implementation for unix?
In general all these conditional compilation in process handling is hard to maintain, because we need to compile with different features to be sure it will compile... But it was already done like this before, so we can leave it as is if it would require to much work.